src/components/tools/ruler/RulerWidget2D.vue

```diff
@@ -162,36 +160,24 @@ export default defineComponent({
 
     // --- handle pick visibility --- //
 
-    const usePointVisibility = (
-      pointState: Ref<vtkRulerWidgetPointState | undefined>
-    ) => {
-      const visible = ref(false);
-      const updateVisibility = () => {
-        if (!pointState.value) return;
-        visible.value = pointState.value.getVisible();
-      };
-
-      onVTKEvent(pointState, 'onModified', () => updateVisibility());
-
-      watchOnce(pointState, () => updateVisibility());
-
-      return visible;
-    };
+    const visibleStates = reactive({
+      firstPoint: false,
+      secondPoint: false,
+    });
 
-    const firstPointVisible = usePointVisibility(
-      computed(() => widget.value?.getWidgetState().getFirstPoint())
-    );
-    const secondPointVisible = usePointVisibility(
-      computed(() => widget.value?.getWidgetState().getSecondPoint())
-    );
+    const widgetState = widgetFactory.getWidgetState();
+    onVTKEvent(widgetFactory.getWidgetState(), 'onModified', () => {
+      visibleStates.firstPoint = widgetState.getFirstPoint().getVisible();
+      visibleStates.secondPoint = widgetState.getSecondPoint().getVisible();
+    });
```

Comment:
Rulers/Rectangles are not showing up upon deserialization.  This bring'em back:
```
    const widgetState = widgetFactory.getWidgetState();

    const visibleStates = reactive({
      firstPoint: widgetState.getFirstPoint().getVisible(),
      secondPoint: widgetState.getSecondPoint().getVisible(),
    });
```

Reply:
Good catch! Fixed.